storages/backends/gcloud.py

```diff
@@ -318,19 +328,24 @@ def url(self, name, parameters=None):
                 storage_base_url=self.custom_endpoint,
                 quoted_name=_quote(name, safe=b"/~"),
             )
+        elif not self.custom_endpoint:
```

Comment:
We don't need the new method or this chain, might be better written as follows. Before that, I am wondering if we need to always include the new parameters such as "access_token" in all of the requests in all cases?

```python
else:
    params = {
            "service_account_email": self.credentials.service_account_email,
            "access_token": self.credentials.token,
            "credentials": self.credentials,
            "expiration": self.expiration,
    }
    if self.custom_endpoint:
        params["api_access_endpoint"] = self.custom_endpoint
    return blob.generate_signed_url(**params)
```

Reply:
I see from the docs that if we don't pass the credentials the client tries to obtain it anyway which means we probably should pass it in. The other two I'm less sure about though since we've already got the credentials there.

Reply:
Sure I have updated. Yeah I think we should always pass it if we can.